bl2/ext/mcuboot/bl2_main.c

```diff
@@ -145,26 +145,36 @@ static void do_boot(struct boot_rsp *rsp)
      */
     __set_MSPLIM(0);
 #endif
-
+#if BL2_TEMP_DIS
     __set_MSP(vt->msp);
     __DSB();
     __ISB();
 
     boot_jump_to_next_image(vt->reset);
+#else
+    SCB->VTOR = ((int) (&(vt->msp)) & 0x1FFFFF80);
```

Comment:
Check other vendors implementation

Reply:
The other vendors initialize SCB->VTOR in their SystemInit() function. 
That register is supposed to be in an undefined state out of reset according to this page: https://developer.arm.com/documentation/100235/0004/the-cortex-m33-peripherals/system-control-block/system-control-block-registers-summary?lang=en
So, maybe our SystemInit should be updated

Reply:
Updated.